src/authentication/components/PermissionGuard.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { Fragment, FunctionComponent, ReactNode } from 'react';
+
+import { useSlot } from '../../shared/hooks/useSlot'; // TODO replace this with avo2-components useSlot once exported
+import { Permissions, PermissionService } from '../helpers/permission-service';
+import { PermissionGuardFail, PermissionGuardPass } from './PermissionGuard.slots';
+
+export interface PermissionGuardProps {
+	children: ReactNode;
+	permissions: Permissions;
+}
+
+const PermissionGuard: FunctionComponent<PermissionGuardProps> = ({ children, permissions }) => {
+	const childrenIfPassed = useSlot(PermissionGuardPass, children);
```

Comment:
Aangezien er niets gewrapped wordt moet je niet persé slots gaan gebruiken maar kan je gwn props gebruiken voor te renderen.

Reply:
De ui die je enkel wil tonen als een user een bepaalde permissie heeft wordt gewrapt
in het voorbeeld is dit de edit button van de collectie

Die disabled edit button in the PermissionGuardFail slot is enkel om te tonen dat je ook ui kan tonen als een user een bepaalde permissie niet heeft.

Reply:
Ik heb het mss niet goed verwoord, je kan dit toch ook gwn via een renderPass en renderFail prop doen? Dan moet je niet telkens 3 imports doen voor 1 iets te renderen.
Maar met slots gaat het ook natuurlijk.

Reply:
ah, je bedoeld 2 render functies meegeven voor success en fail? eg:
```
<PermissionGuard renderPass="renderButton" renderFail="renderDisabledButton">
``` 
Ja dat zou kunnen.
Je kan de component ook gebruiken zonder slots, dan worden de children automatisch gebruikt voor de PermissionGuardPass